app/src/main/java/org/oppia/android/app/devoptions/AppRestartDialogFragmentPresenter.kt

```diff
@@ -22,14 +22,15 @@ class AppRestartDialogFragmentPresenter @Inject constructor(
     )
     binding.lifecycleOwner = fragment
 
+    val appRestartInterface = fragment.parentFragment as AppRestartListener
     val dialog = AlertDialog.Builder(activity, R.style.OppiaAlertDialogTheme)
```

Comment:
```suggestion
    val dialog = AlertDialog.Builder(fragment.activity, R.style.OppiaAlertDialogTheme)
```

Then you can remove `private val activity: AppCompatActivity,`  from the constructor and its import

Reply:
`activity` is also used in :
```kotlin
 val binding = AppRestartDialogFragmentBinding.inflate(
      activity.layoutInflater,
      /* parent= */ null,
      /* attachToRoot= */ false
    )
```
and there it should not be `nullable` so utilising `activity` over `fragment.activity`.

Reply:
Oh I planned to delete the comment but forgot O_o. Sorry!